bson/doc.go

```diff
@@ -19,24 +20,90 @@
+// 		7. BSON array unmarshals to a bson.A.
+// 		8. BSON ObjectId unmarshals to a primitive.ObjectID.
+// 		9. BSON datetime unmarshals to a primitive.Datetime.
+// 		10. BSON binary unmarshals to a primitive.Binary.
+// 		11. BSON regular expression unmarshals to a primitive.Regex.
+// 		12. BSON JavaScript unmarshals to a primitive.JavaScript.
+// 		13. BSON code with scope unmarshals to a primitive.CodeWithScope.
+// 		14. BSON timestamp unmarshals to an primitive.Timestamp.
+// 		15. BSON 128-bit decimal unmarshals to an primitive.Decimal128.
+// 		16. BSON min key unmarshals to an primitive.MinKey.
+// 		17. BSON max key unmarshals to an primitive.MaxKey.
+// 		18. BSON undefined unmarshals to a primitive.Undefined.
+// 		19. BSON null unmarshals to a primitive.Null.
+// 		20. BSON DBPointer unmarshals to a primitive.DBPointer.
+// 		21. BSON symbol unmarshals to a primitive.Symbol.
 //
-// Example:
-// 		b, err := bson.Marshal(bson.D{{"foo", "bar"}})
-// 		if err != nil { return err }
-// 		var fooer struct {
-// 			Foo string
-// 		}
-// 		err = bson.Unmarshal(b, &fooer)
-// 		if err != nil { return err }
-// 		// do something with fooer...
+// The above mappings also apply when marshalling a D or M to BSON. Some other useful marshalling mappings are:
+//
+//       1. time.Time marshals to a BSON datetime.
+//       2. int8, int16, and int32 marshal to a BSON int32.
+//       3. int marshals to a BSON int32 if the value is between math.MinInt32 and math.MaxInt32, inclusive, and a BSON int64
+//       otherwise.
+//       4. int64 marshals to BSON int64.
+//       5. uint8 and uint16 marshal to a BSON int32.
+//       6. uint, uint32, and uint64 marshal to a BSON int32 if the value is between math.MinInt32 and math.MaxInt32,
+//       inclusive, and BSON int64 otherwise.
+//
+// Structs
+//
+// Structs can be marshalled/unmarshalled to/from BSON. When transforming structs to/from BSON, the following rules
+// apply:
+//
+//     1. Only exported fields in structs will be marshalled or unmarshalled.
+//
+//     2. When marshalling a struct, each field will be lowercased to generate the key for the corresponding BSON element.
+//     For example, a struct field named "Foo" will generate key "foo". This can be overriden via a struct tag (e.g.
+//     `bson:"fooField"` to generate key "fooField" instead).
+//
+//     3. An embedded struct field is marshalled as a subdocument. The key will be the lowercased name of the field's type.
+//
+//     4. A pointer field is marshalled as the underlying type if the pointer is non-nil. If the pointer is nil, it is
+//     marshalled as a BSON null value.
+//
+// The following struct tags can be used to configure behavior:
+//
+//     1. omitempty: If the omitempty struct tag is specified on a field, the field will not be marshalled if it is set to
+//     the zero value. By default, omitempty is not relevant for struct fields, which will be serialized as an embedded
```

Comment:
In default mode omitempty still applies for structs that implement IsZero()

Reply:
done.